blocks/snippets.py

```diff
@@ -0,0 +1,14 @@
+"""Routines to build frequently used Theano expressions."""
+from theano import tensor
+
+
+def L2_norm(tensors):
+    """Computes the total L2 norm of a set of tensors.
+
+    Parameters
+    ----------
+    tensors : iterable of :class:`~theano.Variable`
+        The tensors.
+
+    """
+    return tensor.sqrt(sum((t ** 2).sum() for t in tensors))
```

Comment:
I think we should filter it out: I hate finding errors in hastily entered snippets like this. The way you do it above is shorter, but to be honest I do not understand how it works and `tensor.sqr?` gives me quite a piece of documentation to read, so I would leave the code I currently have which I believe everyone can read.

I do not like `blocks.expressions` because it is too general. `blocks.theano_snippets` would work best for me.